{"name": "test_get[dfmfdsdfdddfsffddf]", "status": "broken", "statusDetails": {"message": "TypeError: can only concatenate str (not \"int\") to str", "trace": "self = <api.department.Department object at 0x00000223E0B2E128>\nsteps = [{'method': 'tag.get', 'userid': '${userid}'}, {'assertion': 0, 'path': '$..errcode'}]\n\n    def api_steps(self, steps: list):\n        for step in steps:\n            raw = yaml.dump(step)\n            for key, value in self.params.items():\n                raw = raw.replace(f'${{{key}}}', repr(value))\n            step = yaml.safe_load(raw)\n            if isinstance(step, dict):\n                if 'method' in step.keys():\n                    Get_log().log_info('正在测试'+ step['method'])\n                    allure.attach(step['method']+'测试')\n                    method = step['method'].split('.')[-1]\n                    getattr(self, method)(**step)\n                if \"path\" in step.keys():\n                    self.code = getattr(self, 'jsonpath')(**step)\n                    if isinstance(self.code,int):\n                        self.code =str(self.code)\n                    print(self.code)\n                    print('111111111111111111111')\n                if 'assertion' in step.keys():\n                    try:\n>                       assert self.code == step['assertion']\nE                       AssertionError\n\n..\\api\\baseapi.py:73: AssertionError\n\nDuring handling of the above exception, another exception occurred:\n\nself = <testcase.test_department.TestDepartment object at 0x00000223E17C67F0>\nuserid = 'dfmfdsdfdddfsffddf'\n\n    @allure.step('测试')\n    @pytest.mark.parametrize('userid',['dfmfdsdfdddfsffddf','djkfhdsjfdss'])\n    def test_get(self,userid):\n        self.department.params['userid'] =userid\n>       self.department.api_steps(self.data)\n\ntest_department.py:16: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <api.department.Department object at 0x00000223E0B2E128>\nsteps = [{'method': 'tag.get', 'userid': '${userid}'}, {'assertion': 0, 'path': '$..errcode'}]\n\n    def api_steps(self, steps: list):\n        for step in steps:\n            raw = yaml.dump(step)\n            for key, value in self.params.items():\n                raw = raw.replace(f'${{{key}}}', repr(value))\n            step = yaml.safe_load(raw)\n            if isinstance(step, dict):\n                if 'method' in step.keys():\n                    Get_log().log_info('正在测试'+ step['method'])\n                    allure.attach(step['method']+'测试')\n                    method = step['method'].split('.')[-1]\n                    getattr(self, method)(**step)\n                if \"path\" in step.keys():\n                    self.code = getattr(self, 'jsonpath')(**step)\n                    if isinstance(self.code,int):\n                        self.code =str(self.code)\n                    print(self.code)\n                    print('111111111111111111111')\n                if 'assertion' in step.keys():\n                    try:\n                        assert self.code == step['assertion']\n                    except Exception as e:\n    \n>                       Get_log().log_error(self.code + '！=' + step['assertion'])\nE                       TypeError: can only concatenate str (not \"int\") to str\n\n..\\api\\baseapi.py:76: TypeError"}, "steps": [{"name": "测试", "status": "broken", "statusDetails": {"message": "TypeError: can only concatenate str (not \"int\") to str\n", "trace": "  File \"D:\\python\\lib\\site-packages\\allure_commons\\_allure.py\", line 172, in impl\n    return func(*a, **kw)\n  File \"H:\\练习\\test_reques_wework\\testcase\\test_department.py\", line 16, in test_get\n    self.department.api_steps(self.data)\n  File \"H:\\练习\\test_reques_wework\\api\\baseapi.py\", line 76, in api_steps\n    Get_log().log_error(self.code + '！=' + step['assertion'])\n"}, "attachments": [{"source": "8921fdf5-d988-40ab-af1b-d5548c0d2b6f-attachment.attach"}], "parameters": [{"name": "userid", "value": "'dfmfdsdfdddfsffddf'"}], "start": 1634410152979, "stop": 1634410155475}], "attachments": [{"name": "log", "source": "b79c70f8-76a4-42ed-909c-6de679a91ed8-attachment.txt", "type": "text/plain"}], "parameters": [{"name": "userid", "value": "'dfmfdsdfdddfsffddf'"}], "start": 1634410152979, "stop": 1634410155475, "uuid": "35b2a943-2b1f-41d5-b044-9121c843d3a6", "historyId": "a058b37d5c12e354d9b3efef1ea538ee", "testCaseId": "dcccf3236d29225883a842a7b88face6", "fullName": "testcase.test_department.TestDepartment#test_get", "labels": [{"name": "parentSuite", "value": "testcase"}, {"name": "suite", "value": "test_department"}, {"name": "subSuite", "value": "TestDepartment"}, {"name": "host", "value": "PC-202008151337"}, {"name": "thread", "value": "1552-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "testcase.test_department"}]}